### ISSUE TITLE
New pbkdf2-hmac-md5.h assumes a particular OpenSSL version

### ISSUE THREAD

**Issue by magnumripper** [magnumripper/JohnTheRipper#1447]
Works fine here, but CircleCI fails
```
In file included from pbkdf2-hmac-md5_fmt_plug.c:26:0:
pbkdf2_hmac_md5.h: In function '_pbkdf2_md5_sse_load_hmac':
pbkdf2_hmac_md5.h:131:21: error: 'MD5_CBLOCK' undeclared (first use in this function)
  unsigned char ipad[MD5_CBLOCK], opad[MD5_CBLOCK], k0[MD5_DIGEST_LENGTH];
                     ^
pbkdf2_hmac_md5.h:131:21: note: each undeclared identifier is reported only once for each function it appears in
pbkdf2_hmac_md5.h:131:55: error: 'MD5_DIGEST_LENGTH' undeclared (first use in this function)
  unsigned char ipad[MD5_CBLOCK], opad[MD5_CBLOCK], k0[MD5_DIGEST_LENGTH];
                                                       ^
pbkdf2_hmac_md5.h:141:22: warning: passing argument 2 of 'john_MD5_Update' discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
    MD5_Update( &ctx, K[j], KL[j]);
                      ^
In file included from pbkdf2_hmac_md5.h:11:0,
                 from pbkdf2-hmac-md5_fmt_plug.c:26:
md5.h:31:20: note: expected 'void *' but argument is of type 'const unsigned char *'
 #define MD5_Update john_MD5_Update
                    ^
md5.h:43:13: note: in expansion of macro 'MD5_Update'
 extern void MD5_Update(MD5_CTX *ctx, void *data, unsigned long size);
             ^
In file included from pbkdf2-hmac-md5_fmt_plug.c:26:0:
pbkdf2_hmac_md5.h:131:52: warning: unused variable 'k0' [-Wunused-variable]
  unsigned char ipad[MD5_CBLOCK], opad[MD5_CBLOCK], k0[MD5_DIGEST_LENGTH];
                                                    ^
pbkdf2_hmac_md5.h:131:34: warning: unused variable 'opad' [-Wunused-variable]
  unsigned char ipad[MD5_CBLOCK], opad[MD5_CBLOCK], k0[MD5_DIGEST_LENGTH];
                                  ^
pbkdf2_hmac_md5.h:131:16: warning: unused variable 'ipad' [-Wunused-variable]
  unsigned char ipad[MD5_CBLOCK], opad[MD5_CBLOCK], k0[MD5_DIGEST_LENGTH];
                ^
pbkdf2_hmac_md5.h: In function 'pbkdf2_md5_sse':
pbkdf2_hmac_md5.h:161:25: error: 'MD5_DIGEST_LENGTH' undeclared (first use in this function)
  unsigned char tmp_hash[MD5_DIGEST_LENGTH];
                         ^
pbkdf2_hmac_md5.h:199:29: error: 'MD5_CTX {aka struct <anonymous>}' has no member named 'A'
   ptmp[0]          = ipad[j].A;
                             ^
pbkdf2_hmac_md5.h:200:33: error: 'MD5_CTX {aka struct <anonymous>}' has no member named 'B'
   ptmp[SIMD_COEF_32]   = ipad[j].B;
                                 ^
pbkdf2_hmac_md5.h:201:33: error: 'MD5_CTX {aka struct <anonymous>}' has no member named 'C'
   ptmp[SIMD_COEF_32*2] = ipad[j].C;
                                 ^
pbkdf2_hmac_md5.h:202:33: error: 'MD5_CTX {aka struct <anonymous>}' has no member named 'D'
   ptmp[SIMD_COEF_32*3] = ipad[j].D;
                                 ^
pbkdf2_hmac_md5.h:205:29: error: 'MD5_CTX {aka struct <anonymous>}' has no member named 'A'
   ptmp[0]          = opad[j].A;
                             ^
pbkdf2_hmac_md5.h:206:33: error: 'MD5_CTX {aka struct <anonymous>}' has no member named 'B'
   ptmp[SIMD_COEF_32]   = opad[j].B;
                                 ^
pbkdf2_hmac_md5.h:207:33: error: 'MD5_CTX {aka struct <anonymous>}' has no member named 'C'
   ptmp[SIMD_COEF_32*2] = opad[j].C;
                                 ^
pbkdf2_hmac_md5.h:208:33: error: 'MD5_CTX {aka struct <anonymous>}' has no member named 'D'
   ptmp[SIMD_COEF_32*3] = opad[j].D;
                                 ^
pbkdf2_hmac_md5.h:217:21: warning: passing argument 2 of 'john_MD5_Update' discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
    MD5_Update(&ctx, S, SL);
                     ^
In file included from pbkdf2_hmac_md5.h:11:0,
                 from pbkdf2-hmac-md5_fmt_plug.c:26:
md5.h:31:20: note: expected 'void *' but argument is of type 'const unsigned char *'
 #define MD5_Update john_MD5_Update
                    ^
md5.h:43:13: note: in expansion of macro 'MD5_Update'
 extern void MD5_Update(MD5_CTX *ctx, void *data, unsigned long size);
             ^
In file included from pbkdf2-hmac-md5_fmt_plug.c:26:0:
pbkdf2_hmac_md5.h:237:40: error: 'MD5_CTX {aka struct <anonymous>}' has no member named 'A'
    ptmp[0]           = dgst[j][0] = ctx.A;
                                        ^
pbkdf2_hmac_md5.h:238:44: error: 'MD5_CTX {aka struct <anonymous>}' has no member named 'B'
    ptmp[SIMD_COEF_32]    = dgst[j][1] = ctx.B;
                                            ^
pbkdf2_hmac_md5.h:239:44: error: 'MD5_CTX {aka struct <anonymous>}' has no member named 'C'
    ptmp[SIMD_COEF_32*2]  = dgst[j][2] = ctx.C;
                                            ^
pbkdf2_hmac_md5.h:240:44: error: 'MD5_CTX {aka struct <anonymous>}' has no member named 'D'
    ptmp[SIMD_COEF_32*3]  = dgst[j][3] = ctx.D;
                                            ^
pbkdf2_hmac_md5.h:172:42: warning: unused variable 'sse_crypt2' [-Wunused-variable]
  JTR_ALIGN(MEM_ALIGN_SIMD) unsigned char sse_crypt2[MD5_DIGEST_LENGTH*SSE_GROUP_SZ_MD5];
                                          ^
pbkdf2_hmac_md5.h:171:42: warning: unused variable 'sse_crypt1' [-Wunused-variable]
  JTR_ALIGN(MEM_ALIGN_SIMD) unsigned char sse_crypt1[MD5_DIGEST_LENGTH*SSE_GROUP_SZ_MD5];
                                          ^
pbkdf2_hmac_md5.h:164:15: warning: unused variable 'dgst' [-Wunused-variable]
  ARCH_WORD_32 dgst[SSE_GROUP_SZ_MD5][MD5_DIGEST_LENGTH/sizeof(ARCH_WORD_32)];
               ^
pbkdf2_hmac_md5.h:161:16: warning: unused variable 'tmp_hash' [-Wunused-variable]
  unsigned char tmp_hash[MD5_DIGEST_LENGTH];
                ^
Makefile:1335: recipe for target 'pbkdf2-hmac-md5_fmt_plug.o' failed
make[1]: *** [pbkdf2-hmac-md5_fmt_plug.o] Error 1
make[1]: *** Waiting for unfinished jobs....
Makefile:181: recipe for target 'default' failed
make: *** [default] Error 2
```
Not quite sure how to proceed with the context struct members. How do we get away with SHA1 and the likes? I guess they happen to differ less between crypto libs.

**Comment by magnumripper** [comment]
So we may need a whole stack of #ifdef's :cry:

**Comment by jfoug** [comment]
I hope not.  I build just fine on my Fedora.

I think it is not loading openssl.  I am debugging now.

**Comment by magnumripper** [comment]
I slept on it FTW, fixed in d0934d70. A couple harmless warnings remain after 399c04cd3 but I will fix them soon.
